spark/src/main/scala/org/apache/spark/sql/delta/ResolveDeltaMergeInto.scala

```diff
@@ -278,14 +278,25 @@ object ResolveDeltaMergeInto {
         })
 
       val migrationSchema = filterSchema(source.schema, Seq.empty)
+      val allowTypeWidening = EliminateSubqueryAliases(target) match {
+        case DeltaFullTable(_, index) =>
+          TypeWidening.isEnabled(
+            index.snapshotAtAnalysis.protocol,
+            index.snapshotAtAnalysis.metadata
+          )
+        case o => throw DeltaErrors.notADeltaSourceException("MERGE", Some(o))
```

Comment:
nit: o -> other
Also, this case is probably unreachable code. But outside of that it's mixing too much different concerns, one is a table feature and the other verifying a target table is a Delta table.

Reply:
Updated.
We have a similar pattern later on in `PreprocessTableMerge` and I don't know if there are corner cases that could actually hit it - e.g. some temp view with a filter. To be safe Im' throwing the same error as in `PreprocessTableMerge` so that the behavior doesn't change when schema evolution is enabled vs. disabled

Reply:
Actually changed matching to make it more straight to the point: find the Delta relation (which really should always be there) if we don't find it then consider that type widening is disabled, which is likely safer since something's obviously wrong already